core/storage/question/gae_models.py

```diff
@@ -132,3 +169,108 @@ def create(
+        """Returns ID of the question commit log entry model.
+
+        Args:
+            question_id: str. The question id whose states are mapped.
+            question_version: int. The version of the question.
+
+        Returns:
+            str. A string containing question ID and
+                question version.
+        """
+        return 'question-%s-%s' % (question_id, question_version)
+
+
+class QuestionSummaryModel(base_models.BaseModel):
+    """Summary model for an Oppia question.
+
+    This should be used whenever the content blob of the question is not
+    needed (e.g. in search results, etc).
+
+    A QuestionSummaryModel instance stores the following information:
+
+    creator_id, language_code, status, question_model_last_updated,
+    question_model_created_on, question_data.
+
+    The key of each instance is the question id.
+    """
+    # The user ID of the creator of the question.
+    creator_id = ndb.StringProperty(required=True)
+    # The ISO 639-1 code for the language this question is written in.
+    language_code = ndb.StringProperty(required=True, indexed=True)
+    # The status of the question. It can be 'private', 'approved',
+    # 'rejected' and 'pending'.
+    status = ndb.StringProperty(
+        default=feconf.ACTIVITY_STATUS_PRIVATE, indexed=True,
+        choices=[
+            feconf.ACTIVITY_STATUS_PRIVATE,
+            feconf.QUESTION_STATUS_APPROVED,
+            feconf.QUESTION_STATUS_REJECTED,
+            feconf.QUESTION_STATUS_PENDING
+        ]
+    )
+    # Time when the question model was last updated (not to be
+    # confused with last_updated, which is the time when the
+    # question *summary* model was last updated).
+    question_model_last_updated = ndb.DateTimeProperty(indexed=True)
+    # Time when the question model was created (not to be confused
+    # with created_on, which is the time when the question *summary*
+    # model was created).
+    question_model_created_on = ndb.DateTimeProperty(indexed=True)
+    # A dict representing the question data.
+    question_html_data = ndb.StringProperty(indexed=False, required=True)
+
+    @classmethod
+    def get_by_creator_id(cls, creator_id):
+        """Gets QuestionSummaryModel by creator_id.
+
+        Args:
+            creator_id: str. The user ID of the creator of the question.
+
+        Returns:
+            QuestionSummaryModel. The summary model of the question.
+        """
+        return QuestionSummaryModel.query().filter(
+            cls.creator_id == creator_id).fetch()
+
+
+class QuestionRightsSnapshotMetadataModel(
+        base_models.BaseSnapshotMetadataModel):
+    """Storage model for the metadata for a question rights snapshot."""
+    pass
+
+
+class QuestionRightsSnapshotContentModel(base_models.BaseSnapshotContentModel):
+    """Storage model for the content of a question rights snapshot."""
+    pass
+
+
+class QuestionRightsModel(base_models.VersionedModel):
```

Comment:
Ditto about whether you plan to use question rights anywhere?

Reply:
Removed this model until further discussion.